feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/novacard/overview/NovaCardViewModel.kt

```diff
@@ -40,12 +42,25 @@ class NovaCardViewModel(
     }
 
     fun openTopUp(amount: BigDecimal, address: String) = launch {
-        assetsRouter.openTopUpCard(
-            TopUpCardPayload(
-                amount = amount,
-                address = address,
-                asset = setupCardConfig.first().spendToken.toAssetPayload()
-            )
+        val payload = TopUpCardPayload(
+            amount = amount,
+            address = address,
+            asset = setupCardConfig.first().spendToken.toAssetPayload()
         )
+
+        assetsRouter.openTopUpCard(payload)
+    }
+
+    fun onCardCreated() {
+        // No need to open/close timer dialog if card is already active
+        if (novaCardInteractor.isNovaCardStateActive()) return
+
+        novaCardInteractor.setNovaCardState(true)
+
+        closeTimerBottomSheet()
```

Comment:
Typically we dont do it so a screen closes another screen since it might lead to harder understanding of the flow
Given that Timer Bottom Sheet is a seaparate screen with its own view model, it could close itself upon noticing a nova card state being true